contrib/test.sh

```diff
@@ -1,11 +1,41 @@
 #!/bin/bash
 set -e
 
-# Run tests for the Rust project
-echo "Running Rust tests..."
-cargo test --package payjoin --verbose --all-features --lib
-cargo test --package payjoin --verbose --features=send,receive --test integration
-cargo test --package payjoin --verbose --no-default-features --features=send,receive,danger-local-https,v2 --test integration
-cargo test --package payjoin-cli --verbose --no-default-features --features=danger-local-https,v2 --test e2e
-cargo test --package payjoin-cli --verbose --features=danger-local-https
+DEPS="recent minimal"
 
+for dep in $DEPS
+do
+    cargo --version
+    rustc --version
+
+    # Some tests require certain toolchain types.
+    NIGHTLY=false
+    STABLE=true
+    if cargo --version | grep nightly; then
+        STABLE=false
+        NIGHTLY=true
+    fi
+    if cargo --version | grep beta; then
+        STABLE=false
+    fi
+    # Pin dependencies as required if we are using MSRV toolchain.
+    if cargo --version | grep "1\.63"; then
+        cargo update -p cc --precise 1.0.105
+        cargo update -p clap_lex --precise 0.3.0
+        cargo update -p regex --precise 1.9.6
+        cargo update -p reqwest --precise 0.12.4
+        cargo update -p time --precise 0.3.20
+        cargo update -p tokio --precise 1.38.1
+        cargo update -p tokio-util --precise 0.7.11
+        cargo update -p url --precise 2.5.0
+        cargo update -p which --precise 4.4.0
+        cargo update -p zstd-sys --precise 2.0.8+zstd.1.5.5
+    fi
```

Comment:
Isn't the whole point of `Cargo-minimal.lock` that we don't have to pin these MSRV versions manually?

Reply:
Hm, I saw that rust-bitcoin was still pinning like this in https://github.com/rust-bitcoin/rust-bitcoin/pull/1764. The idea was that the Cargo-minimal.lock let downstream users get the pins without manually entering them, having them specified in one place instead of in both CI yaml and in the readme. But we could remove this entirely, we'd just lose some track of exactly what has been updated, but I suppose that *would* make the lock file the one true dependency reference, so I've removed these update lines